nouveau/src/main/java/org/apache/couchdb/nouveau/core/IndexManager.java

```diff
@@ -172,6 +155,21 @@ public <R> R with(final String name, final IndexFunction<Index, R> indexFun)
         }
     }
 
+    private void evictIfOverCapacity() throws IOException, InterruptedException {
+        while (true) {
+            final String candidate;
+            synchronized (cache) {
+                candidate = cache.size() > maxIndexesOpen
+                        ? cache.entrySet().iterator().next().getKey()
+                        : null;
```

Comment:
Would it be simpler to return directly from the synchronized block when `cache.size() > maxIndexesOpen`. Might save the extra `if (candidate == null) ...` lines?

Reply:
sure